src/test/java/org/opensearch/sdk/TestNamedWriteableRegistryApi.java

```diff
@@ -0,0 +1,171 @@
+            new TransportAddress(InetAddress.getByName("localhost"), 9876),
+            emptyMap(),
+            emptySet(),
+            Version.CURRENT
+        );
+        OpenSearchRequest request = new OpenSearchRequest(OpenSearchRequestType.REQUEST_OPENSEARCH_NAMED_WRITEABLE_REGISTRY);
+        NamedWriteableRegistryResponse response = namedWriteableRegistryApi.handleNamedWriteableRegistryRequest(request);
+
+        // Verify that the api processes named writeable registry entries successfully within the response
+        assertEquals(response.getRegistry().size(), 1);
+        assertEquals(response.getRegistry().get(Example.NAME), Example.class);
+    }
+
+    @Test
+    public void testNamedWriteableRegistryParseRequest() throws UnknownHostException {
+        DiscoveryNode sourceNode = new DiscoveryNode(
+            "test_node",
+            new TransportAddress(InetAddress.getByName("localhost"), 9876),
+            emptyMap(),
+            emptySet(),
+            Version.CURRENT
+        );
+
+        // convert stream output into byte array
+        byte[] context = null;
+        ByteArrayOutputStream buf = new ByteArrayOutputStream();
+        OutputStreamStreamOutput out = new OutputStreamStreamOutput(buf);
+        try {
+
+            // named writeables are always preceded by the writeable name within the streaminput
+            out.writeString(Example.NAME);
+            out.writeString("test_message");
+            context = buf.toByteArray();
+
+            NamedWriteableRegistryParseRequest request = new NamedWriteableRegistryParseRequest(Example.class.getName(), context);
+            BooleanResponse response = namedWriteableRegistryApi.handleNamedWriteableRegistryParseRequest(request);
+
+            // verify that byte array deserialization was successful
+            assertEquals(response.getStatus(), true);
+        } catch (IOException e) {}
+    }
+
+    @Test
+    public void testInvalidNamedWriteableRegistryParseRequest() throws UnknownHostException {
+        DiscoveryNode sourceNode = new DiscoveryNode(
+            "test_node",
+            new TransportAddress(InetAddress.getByName("localhost"), 9876),
+            emptyMap(),
+            emptySet(),
+            Version.CURRENT
+        );
+
+        // convert stream output into byte array
+        byte[] context = null;
+        ByteArrayOutputStream buf = new ByteArrayOutputStream();
+        OutputStreamStreamOutput out = new OutputStreamStreamOutput(buf);
+        try {
+
+            // write the invalid name to the stream output
+            out.writeString(Example.INVALID_NAME);
+            out.writeString("test_message");
+            context = buf.toByteArray();
+
+            try {
+                NamedWriteableRegistryParseRequest request = new NamedWriteableRegistryParseRequest(Example.class.getName(), context);
+                BooleanResponse response = namedWriteableRegistryApi.handleNamedWriteableRegistryParseRequest(request);
```

Comment:
Where are we testing the request in the test? Am I missing something here?

Reply:
The parse request will fail if the named writeable is not registered within the NamedWriteableRegistry of the API. Named writeables are preceded by their writeable name within the byte array sent to the SDK, and this name is read first in order to identify the correct reader within the registry to apply to the StreamInput. Here we write an invalid writeable name to the StreamOutput, which would trigger an IllegalArguementException in the SDK since this name is not registered

Reply:
Yeah, so test the exception

Reply:
I'll change the name of the test so that it's purpose is more clear